spec_util/meld.go

```diff
@@ -184,13 +187,8 @@ func MeldData(dst, src *pb.Data) (retErr error) {
 	case *pb.Data_Optional:
 		switch opt := v.Optional.Value.(type) {
 		case *pb.Optional_Data:
-			// Check whether src == dst with optional removed.
-			// E.g. meld(optional<string>, string) => optional<string>
-			if dataEqual(opt.Data, src) {
-				// Do nothing - src is just non-optional verison of dst.
-				return nil
-			}
-			return recordConflict(dst, src)
+			// Meld src with the non-optional version of dst.
+			return MeldData(opt.Data, src)
```

Comment:
Do you have to call `makeOptional` on the result here, or otherwise the optional will be discarded?

Reply:
No, because we are calling `MeldData` on the thing *inside* of the optional here, so the outer optional is retained.

Reply:
Ah, right, because this overwrites `dst` rather than producing a new data object.  Got it.